packages/babel-helper-replace-supers/src/index.js

```diff
@@ -42,8 +42,10 @@ export function skipAllButComputedKey(path: NodePath) {
 }
 
 // environmentVisitor should be used when traversing the whole class and not for specific class elements/methods.
+// For perf reasons, the environmentVisitor will be traversed with `{ noScope: true }`, which means `path.scope` is undefined.
+// Avoid using `path.scope` here
 export const environmentVisitor = {
-  TypeAnnotation(path: NodePath) {
+  "StaticBlock|ClassPrivateProperty|TypeAnnotation"(path: NodePath) {
```

Comment:
We need to make sure that `StaticBlock` is not used with older `@babel/traverse` versions also here.

I don't think we can have a 100% save way of doing so, but this would be a good start:
```suggestion
  [`${t.StaticBlock ? "StaticBlock|" : ""}ClassPrivateProperty|TypeAnnotation`](path: NodePath) {
```

This can still fail if, for example, the `@babel/types` dependency of `@babel/helper-replace-supers` is 7.12 and the `@babel/types` dependency of `@babel/core > @babel/traverse` is older, but I can't think of a way of fixing it. (And the solution is just `yarn dedupe`/`npm dedupe`).